articles/creating-a-hangman-game-with-vanilla-javascript/index.md

```diff
@@ -0,0 +1,238 @@
+```javascript
+function updateMistakes() {
+  document.getElementById('mistakes').innerHTML = mistakes;
+}
+```
+Now, the number of wrong guesses will always keep updating by +1.
+The next thing to do now is to check for wins and losses so that the player does not just keep playing endlessly. To do that you run a checkifGameWon(); in the `if` statement of the `handleGuess();` function and a `checkIfGameLost();` in the corresponding `elseif` statement.
+###checkIfGameWon() function
+We are going to start by creating an if statement saying if “wordStatus” is equal to the answer, the keyboard will print out “You won!!!”. Here is the code below:
+```javascript
+function checkIfGameWon() {
+  if (wordStatus === answer) {
+    document.getElementById('keyboard').innerHTML = 'You Won!!!';
+  }
+}
+```
+
+###checkIfGameLost() function
+Here’s the code to check if the game lost.
+```javascript
+function checkIfGameLost() {
+  if (mistakes === maxWrong) {
+    document.getElementById('wordSpotlight').innerHTML = 'The answer was: ' + answer;
+    document.getElementById('keyboard').innerHTML = 'You Lost!!!';
+  }
+}
+```
+In the code above, We set the game to print out “You lost!!!” when the player has reached the maximum number of wrong guesses. We also set the game to print out the correct answer too.
+reset() function
+In this function, we are going to set everything back to default.  Here is how it looks in the code:
+```javascript
+function reset() {
+  mistakes = 0;
+  guessed = [];
+  document.getElementById('hangmanPic').src = 'images/0.jpg';
+
+  randomWord();
+  guessedWord();
+  updateMistakes();
+  generateButtons();
+}
+```
+###updateHangmanPicture () function
+Firstly, we are going to run the `updateHangmanPicture();` in the `elseif` statement of the `handleGuess()` function so that the image updates every time the player gets a letter wrong. Then we create the updateHangmanPicture() function using the code below:
+```javascript
+function updateHangmanPicture() {
+  document.getElementById('hangmanPic').src = 'images/' + mistakes + '.jpg';
+}
+```
+In the code above, we set the images to update each time the player gets an answer wrong continually. I named all six images (0-6).jpg, and we have a maximum number of 6 mistakes to be made. This now conveniently helps us set the pictures to update every time the player gets a letter wrong.
+
+Lastly, we need to make the game accessible to players that wish to navigate with their keyboard. Firstly, you need to define the alphabet keys in your keyboard using the code below:
+```javascript
+Let alphabets=["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"];
+```
+Next, you need to create a function that will link these keys on your keyboard to the game and make them appear as valid guesses as they are clicked by using `document.onkeypress`. Here is the code below:
```

Comment:
Add a single empty line before and after a code block. Please apply this to all the code blocks.